decoder/vaapidecoder_h264.cpp

```diff
@@ -1287,9 +1287,11 @@ void VaapiDecoderH264::fillReference(VAPictureH264* refs, size_t size)
     size_t i = 0;
     DPB::PictureList::iterator it = m_dpb.m_pictures.begin();
 
-    for (; it != m_dpb.m_pictures.end(); i++, it++) {
-        fillVAPictureH264(&refs[i], *it);
-        DEBUG("poc %d, isShortRef %d, isRef %d", (*it)->m_poc,
+    for (; it != m_dpb.m_pictures.end(); it++) {
+        if (!isReference(*it))
```

Comment:
LGTM

Other question,  is it have some issue in the other decoder (e,g HEVC) ?